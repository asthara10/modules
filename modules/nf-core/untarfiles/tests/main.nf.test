nextflow_process {

    name "Test Process UNTARFILES"
    script "../main.nf"
    process "UNTARFILES"

    tag "modules"
    tag "modules_nfcore"
    tag "untarfiles"

    test("test_untarfiles") {


        when {
            process {
                """
                
                input[0] = [
        [id: 'test'],
        file(params.test_data['sarscov2']['genome']['kraken2_tar_gz'], checkIfExists: true)
    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_untarfiles_different_output_path") {
        when {
            process {
                """
                        input[0] = [
        [id: 'test'],
        file(params.test_data['homo_sapiens']['illumina']['test_flowcell'], checkIfExists: true)
    ]


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }
    }
    
    test("test_untarfiles_onlyfiles") {
        when {
            process {
                """
                        input[0] = [
        [id: 'test'],
        file(params.test_data['generic']['tar']['tar_gz'], checkIfExists: true)
    ]


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }
    }
    
}
