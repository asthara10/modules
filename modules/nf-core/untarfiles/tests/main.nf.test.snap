{
    "versions": {
        "content": [
            [
                
            ]
        ],
        "timestamp": "2023-11-24T10:29:07.437537"
    }
}